libs/components/packages/src/schematics/migrations/update-13/icon-name/icon-name.schematic.ts

```diff
@@ -193,36 +198,46 @@ function updateDataViewConfig(
   }
 }
 
-async function updateSourceFiles(tree: Tree): Promise<void> {
+async function updateSourceFiles(
+  tree: Tree,
+  context: SchematicContext,
+): Promise<void> {
   const workspace = await getWorkspace(tree);
 
   workspace.projects.forEach((project) => {
     visitProjectFiles(tree, project.sourceRoot || project.root, (filePath) => {
-      if (filePath.endsWith('.html')) {
-        const doc = parseTemplate(tree.readText(filePath));
-        const recorder = tree.beginUpdate(filePath);
-        COMPONENTS_WITH_ICON.forEach((componentInfo) => {
-          updateIcon(doc, componentInfo, 0, recorder);
-        });
-        tree.commitUpdate(recorder);
-      }
-      if (filePath.endsWith('.ts')) {
-        const source = parseSourceFile(tree, filePath);
-        const recorder = tree.beginUpdate(filePath);
-        if (isImported(source, 'Component', '@angular/core')) {
-          const content = tree.readText(filePath);
-          const templates = getInlineTemplates(source);
-          templates.forEach((template) => {
-            const doc = parseTemplate(
-              content.substring(template.start, template.end),
-            );
-            COMPONENTS_WITH_ICON.forEach((componentInfo) => {
-              updateIcon(doc, componentInfo, template.start, recorder);
-            });
+      try {
+        if (filePath.endsWith('.html')) {
+          const doc = parseTemplate(tree.readText(filePath));
+          const recorder = tree.beginUpdate(filePath);
+          COMPONENTS_WITH_ICON.forEach((componentInfo) => {
+            updateIcon(doc, componentInfo, 0, recorder);
           });
+          tree.commitUpdate(recorder);
         }
-        updateDataViewConfig(source, recorder);
-        tree.commitUpdate(recorder);
+        if (filePath.endsWith('.ts')) {
+          const source = parseSourceFile(tree, filePath);
+          const recorder = tree.beginUpdate(filePath);
+          if (isImported(source, 'Component', '@angular/core')) {
+            const content = tree.readText(filePath);
+            const templates = getInlineTemplates(source);
+            templates.forEach((template) => {
+              const doc = parseTemplate(
+                content.substring(template.start, template.end),
+              );
+              COMPONENTS_WITH_ICON.forEach((componentInfo) => {
+                updateIcon(doc, componentInfo, template.start, recorder);
+              });
+            });
+          }
+          updateDataViewConfig(source, recorder);
+          tree.commitUpdate(recorder);
+        }
+      } catch (error) {
+        // Log the error and continue the migration.
+        context.logger.warn(
+          `Error when updating icons in ${filePath}: ${error}`,
```

Comment:
Do we know what type of file throws this error? Seems odd that a project would build, but then throw a parsing error in the schematic?

Reply:
I was not able to identify a specific issue, it seems odd, and it's within NG's `isImported` where the error was being thrown.